frontend/src/metabase/public/containers/PublicDashboard/PublicDashboard.unit.spec.tsx

```diff
@@ -70,6 +70,16 @@ describe("PublicDashboard", () => {
 
     expect(firstTab).toHaveAttribute("aria-selected", "true");
   });
+
+  it("should work with filter value starting from '0' (metabase#41483)", async () => {
+    // note: as all slugs this is ignored and we only use the id
+    await setup({
+      queryString: "?my-filter-value=01",
+    });
+
+    // should not throw runtime error and render dashboard content
+    expect(screen.getByText(DASHBOARD_TITLE)).toBeInTheDocument();
```

Comment:
This test only verifies that the dashboard is rendered not crashed. But it doesn't assert that the filter value is retained.

I've tried a few iterations to make this unit test work by passing `dashcards` and `parameters` but couldn't do that. Maybe you could try or create an E2E if that's easier. I think it's important to capture that 01 would still be present and used as a filter.

At first I thought this code wasn't fixing the issue because there's also no filter value assertion in the test.